src/pages/RealmExplorer.js

```diff
@@ -3,201 +3,278 @@
 
+const getTableRenderer = realmObjectFields => {
+  const RealmTable = ({ data }) => {
+    if (!realmObjectFields) return null;
+    const TableHeader = getHeaderRenderer(realmObjectFields);
+    const TableRow = getRowRenderer(realmObjectFields);
     return (
-      <View style={[globalStyles.container]}>
-        <SearchBar onChange={this.onSearchChange} placeholder="Table Name" />
-        <SearchBar onChange={this.onFilterChange} placeholder="Filter" />
-        <DataTable
-          style={globalStyles.container}
-          listViewStyle={globalStyles.container}
-          dataSource={dataSource}
-          renderRow={this.renderRow}
-          renderHeader={this.renderHeader}
-        />
-      </View>
+      <VirtualizedList
+        ListHeaderComponent={TableHeader}
+        data={data}
+        getItem={getItem}
+        getItemCount={getItemCount}
+        keyExtractor={keyExtractor}
+        renderItem={TableRow}
+      />
     );
-  }
-}
+  };
+  RealmTable.propTypes = propTypes.RealmTable;
+  return RealmTable;
+};
 
-export default RealmExplorer;
+/**
+ * A page for displaying objects in the local database. Includes search and filtering functionality.
+ *
+ * @prop   {UIDatabase}     database      App wide database.
+ * @state  {string}         searchString  Current search string. Used to update current object.
+ * @state  {string}         filterString  Current filter string. Used to update filtered data.
+ * @state  {Realm.Results}  objectData    Reference to current database object results. Used to roll
+ *                                        back filter state when filter is reset.
+ * @state  {Realm.Results}  filteredData  Reference to current database object results after filter
+ *                                        has been applied. Displayed to the user.
+ */
+export const RealmExplorer = () => {
+  const [state, setState] = useState(getInitialState());
+
+  const onSearchChange = useCallback(
+    newObjectString => setState(prevState => updateSearchData(newObjectString, prevState)),
+    []
+  );
+  const onFilterChange = useCallback(
+    newFilterString => setState(prevState => updateFilteredData(newFilterString, prevState)),
+    []
+  );
+
+  const { searchString, filterString, filteredData } = state;
+
+  const realmObjectFields = REALM_OBJECTS_FIELDS[searchString];
 
-RealmExplorer.propTypes = {
-  // eslint-disable-next-line react/forbid-prop-types
-  database: PropTypes.object.isRequired,
+  const RealmSearchBar = useCallback(getSearchBarRenderer(onSearchChange), [onSearchChange]);
+  const RealmFilterBar = useCallback(getFilterBarRenderer(onFilterChange), [onFilterChange]);
+  const RealmTable = useCallback(getTableRenderer(realmObjectFields), [realmObjectFields]);
+
+  return (
+    <View style={[globalStyles.container]}>
+      <RealmSearchBar searchString={searchString} />
+      <RealmFilterBar filterString={filterString} />
+      <RealmTable data={filteredData} />
```

Comment:
my opinion is that this looks pretty but makes it harder to actually know whats going on at a glance - takes more cognitive load to actually understand the components and requires more thought into memoization and other performance enhancing techniques. I like to use this style when there are conditions on what and when to render something, but not to wrap to many basic components in other components 

```
<RealmSearchBar searchString={searchString}/>
```

with 2? wrapping functions isn't easier to read than:

```
<SearchBar value={filterString} onChangeText={onFilterChange} placeholder="Filter string" />
```

And provides no additional benefit that I can see given that `SearchBar` is already memoized and re-renders only when `value` changes